File: Back.QuizApp/Controllers/Mappers/CategoryMapper.cs
using Ynov.QuizApp.Business.Models;
using Ynov.QuizApp.Controllers.DTOs;

namespace Ynov.QuizApp.Controllers.Mappers;

public class CategoryMapper {
    public CategoryDTO ToDto(Category category) {
        return new CategoryDTO() {
            Id = category.Id,
            Name = category.Name,
            Description = category.Description,
        }; 
    }
}

File: Front.QuizApp/src/app/business/mappers/category.mapper.ts
import {Injectable} from '@angular/core';
import {Category} from '../models/category.model';
import {CategoryDto} from '../dtos/category.dto';

@Injectable({providedIn: 'root'})
export class CategoryMapper {
    public mapCategoryFromApiToModel(apiCategory: CategoryDto): Category {
        return {
            id: apiCategory.id,
            name: apiCategory.name,
            description: apiCategory.description || '',
        };
    }

    public mapCategoryName(categoryName: Category) : string {
        return categoryName.name;
    }
}
